WSFrontend/org.openl.rules.ruleservice.ws/src/org/openl/rules/ruleservice/publish/jaxrs/JAXRSExceptionMapper.java

```diff
@@ -16,6 +21,25 @@ public Response toResponse(Exception e) {
             dto.getType(),
             dto.getDetail() != null ? dto.getDetail().replaceAll("\t", "    ").split(System.lineSeparator()) : null);
 
+        if (ExceptionType.isServerError(dto.getType())) {
+            Throwable t = e.getCause();
+            while (t != null && t.getCause() != t) {
+                if ((t instanceof OpenLException) && t.getMessage() != null) {
+                    break;
+                }
+                t = t.getCause();
+            }
+            Logger logger = LoggerFactory.getLogger(getClass());
+            if (t instanceof OpenLRuntimeException) {
+                logger.error(((OpenLRuntimeException) t).getOriginalMessage(), t);

```

Comment:
Basically, error level means that the application cannot work correctly at all.
Warn level means, that some error happened and the application can work incorrectly.
Info level means, that some error happened and it is expected/sucessfully handled and the application continue to work correctly.

Reply:
I'm logging only SYSTEM, COMPILATION, or RULES_RUNTIME. These exceptions are happening during rules execution and in all cases they mean that rules have critical errors and cannot perform calculation. If these errors happen, client cannot fix request either.